azurerm/internal/services/managedapplication/tests/resource_arm_managed_application_definition_test.go

```diff
@@ -0,0 +1,326 @@
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMManagedApplicationDefinitionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMManagedApplicationDefinition_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMManagedApplicationDefinitionExists(data.ResourceName),
+				),
+			},
+			data.ImportStep("package_file_uri"),
+		},
+	})
+}
+
+func TestAccAzureRMManagedApplicationDefinition_requiresImport(t *testing.T) {
+	if !features.ShouldResourcesBeImported() {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_managed_application_definition", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMManagedApplicationDefinitionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMManagedApplicationDefinition_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMManagedApplicationDefinitionExists(data.ResourceName),
+				),
+			},
+			data.RequiresImportErrorStep(testAccAzureRMManagedApplicationDefinition_requiresImport),
+		},
+	})
+}
+
+func TestAccAzureRMManagedApplicationDefinition_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_managed_application_definition", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMManagedApplicationDefinitionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMManagedApplicationDefinition_complete(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMManagedApplicationDefinitionExists(data.ResourceName),
+				),
+			},
+			data.ImportStep("create_ui_definition", "main_template"),
+		},
+	})
+}
+
+func TestAccAzureRMManagedApplicationDefinition_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_managed_application_definition", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMManagedApplicationDefinitionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMManagedApplicationDefinition_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMManagedApplicationDefinitionExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "display_name", "TestManagedApplicationDefinition"),
+					resource.TestCheckResourceAttr(data.ResourceName, "description", "Test Managed Application Definition"),
+					resource.TestCheckResourceAttr(data.ResourceName, "package_enabled", "false"),
+					resource.TestCheckResourceAttr(data.ResourceName, "tags.%", "0"),
```

Comment:
Also test for attribute `package_file_uri`

Reply:
updated